lib/post_data.js

```diff
@@ -190,10 +193,54 @@ SlackDataPostHandler.prototype.postData = function(data) {
   }
 };
 
+/*
+  MSTeamsPostHandler.
+*/
+function MSTeamsPostHandler(robot, formatter) {
+  this.robot = robot;
+  this.formatter = formatter;
+}
+
+MSTeamsPostHandler.prototype.postData = function(data) {
+  var messages_to_send,
+      // We capture robot here so the `sendMessage` closure captures the
+      // correct `this`
+      robot = this.robot,
+      split_message = utils.splitMessage(data.message);
+
+  if (data.extra && data.extra.botframework) {
+    robot.logger.warning(util.format('The extra.botframework attribute of aliases is not used yet.'));
+  }
+
+  if (split_message.pretext) {
+    var text = this.formatter.formatData(split_message.text);
+    messages_to_send = [split_message.pretext, text];
+  } else {
+    messages_to_send = [this.formatter.formatData(data.message)];
+  }
+
+  // We define a recursive closure that calls itself with the next data to send
+  // after a timeout. This approximates sending synchronous (sequential) HTTP
+  // requests.
+  var sendMessage = function (i) {
+    robot.adapter.send(data.context, messages_to_send[i]);
+
+    if (messages_to_send.length > ++i) {
+      setTimeout(function () { sendMessage(i); }, 300);
```

Comment:
Almost like the IRC days where you were limited to how fast you can send the commands :laughing:

Reply:
It looks like the "worst" case will be a maximum of two messages?

Reply:
Yeah, what's even more annoying is that the underlying JS library takes a `done` callback parameter that is called when the HTTP request resolves, but the `adapter.send` function only accepts a `context` parameter and `messages...` parameter/s. 😆

For now, yes, the worst case will be a `pretext` and a `text` message, but if this adapter gets any more complex (and I expect it to), that will likely change. I completely stole this pattern from the Slack adapter, which is an excellent example of how complex it can get.